syntax = "proto3";

package interact_proto;

service MainService {
    rpc RegisterUser(RegisterRequest) returns (RegisterResponse) {}
    rpc ViewPost(PostEventRequest) returns (PostEventResponse) {}
    rpc LikePost(PostEventRequest) returns (PostEventResponse) {}
    rpc CommentPost(CommentRequest) returns (CommentResponse) {}
    rpc GetComments(GetCommentsRequest) returns (GetCommentsResponse) {}
}

message RegisterRequest {
    int32 user_id = 1;
    string username = 2;
    string registration_date = 3;
}
message RegisterResponse {
    bool success = 1;
    int32 user_id = 2;
    string message = 3;
}

message PostEventRequest {
    int32 post_id = 1;
    int32 user_id = 2;
}
message PostEventResponse {
    bool success = 1;
    string message = 2;
}

message CommentRequest {
    int32 post_id = 1;
    int32 user_id = 2;
    string comment = 3;
}
message CommentResponse {
    bool success = 1;
    string message = 2;
}

message GetCommentsRequest {
    int32 post_id = 1;
    int32 page = 2;
    int32 page_size = 3;
}
message GetCommentsResponse {
    repeated Comment comments = 1;
    int32 comment_cnt = 2;
}
message Comment {
    int32 comment_id = 1;
    int32 user_id = 2;
    string comment = 3;
    string timestamp = 4;
}